tests/ShopBundle/Functional/Model/Cart/Watcher/CartWatcherServiceTest.php

```diff
@@ -4,58 +4,51 @@
 
 use Shopsys\FrameworkBundle\Component\Domain\Domain;
 use Shopsys\FrameworkBundle\DataFixtures\Demo\PricingGroupDataFixture;
+use Shopsys\FrameworkBundle\DataFixtures\Demo\ProductDataFixture;
 use Shopsys\FrameworkBundle\Model\Cart\Cart;
 use Shopsys\FrameworkBundle\Model\Cart\Item\CartItem;
 use Shopsys\FrameworkBundle\Model\Cart\Watcher\CartWatcherService;
 use Shopsys\FrameworkBundle\Model\Customer\CurrentCustomer;
 use Shopsys\FrameworkBundle\Model\Customer\CustomerIdentifier;
 use Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat;
 use Shopsys\FrameworkBundle\Model\Pricing\Vat\VatData;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductManualInputPriceFacade;
 use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceCalculationForUser;
-use Shopsys\FrameworkBundle\Model\Product\ProductCategoryDomainFactory;
 use Shopsys\FrameworkBundle\Model\Product\ProductVisibility;
 use Shopsys\FrameworkBundle\Model\Product\ProductVisibilityRepository;
 use Shopsys\ShopBundle\Model\Product\Product;
 use Shopsys\ShopBundle\Model\Product\ProductDataFactory;
-use Tests\ShopBundle\Test\FunctionalTestCase;
+use Tests\ShopBundle\Test\TransactionFunctionalTestCase;
 
-class CartWatcherServiceTest extends FunctionalTestCase
+class CartWatcherServiceTest extends TransactionFunctionalTestCase
 {
     public function testGetModifiedPriceItemsAndUpdatePrices()
     {
         $customerIdentifier = new CustomerIdentifier('randomString');
-        $productDataFactory = $this->getContainer()->get(ProductDataFactory::class);
 
-        $vatData = new VatData();
-        $vatData->name = 'vat';
-        $vatData->percent = 21;
-        $vat = new Vat($vatData);
-        $productData1 = $productDataFactory->create();
-        $productData1->name = [];
-        $productData1->price = 100;
-        $productData1->vat = $vat;
-        $productData1->priceCalculationType = Product::PRICE_CALCULATION_TYPE_AUTO;
-        $productMock = Product::create($productData1);
+        /** @var \Shopsys\ShopBundle\Model\Product\Product $product */
+        $product = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '1');
 
+        /* @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceCalculationForUser $productPriceCalculationForUser */
```

Comment:
should be "/** @var..." :slightly_smiling_face:

Reply:
however, this is inconsistent throughout multiple classes here, so I would not bother :wink:

Reply:
It would be nice to unify annotations. I've opened an issue to address this problem (#29)